snippets/base/views.py

```diff
@@ -118,7 +118,7 @@ def fetch_render_snippets(request, **kwargs):
                          .filter_by_available())
 
     response = render(request, 'base/fetch_snippets.html', {
-        'snippets': matching_snippets,
+        'snippets_json': json.dumps([s.to_dict() for s in matching_snippets]),
```

Comment:
I see this list comprehension repeated a lot. Consider making a method on the queryset for it: e.g. `matching_snippets.to_json()`.